pages/client-libraries/go.mdx

```diff
@@ -1,130 +1,770 @@
+
+Due to nature of `ExecuteQuery` method, transaction is handled automatically. 
+
+#### Run a read query 
+
+Here is the simple read query:
+
+```go
+result, err := neo4j.ExecuteQuery(ctx, driver, "MATCH (n:Technology{name: 'Memgraph'}) RETURN n", nil, neo4j.EagerResultTransformer, neo4j.ExecuteQueryWithDatabase("memgraph"))
+if err != nil {
+	panic(err)
+} else {
+	fmt.Println(result)
 }
 
-func insertItem(driver neo4j.Driver) (*Item, error) {
-    // Sessions are short-lived, cheap to create and NOT thread safe. Typically create one or more sessions
-    // per request in your web application. Make sure to call Close on the session when done.
-    // For multi-database support, set sessionConfig.DatabaseName to requested database
-    // Session config will default to write mode, if only reads are to be used configure session for
-    // read mode.
-    session := driver.NewSession(neo4j.SessionConfig{})
-    defer session.Close()
-    result, err := session.WriteTransaction(createItemFn)
-    if err != nil {
-        return nil, err
-    }
-    return result.(*Item), nil
+// Print each node as map
+for _, node := range result.Records {
+	fmt.Println(node.AsMap())
 }
+```
+In this query each record contains a node, and you can access it by using `AsMap()` method.
+
+
+#### Running queries with property map
 
-func createItemFn(tx neo4j.Transaction) (interface{}, error) {
-    records, err := tx.Run(
-        "CREATE (a:Greeting) SET a.message = $message RETURN 'Node ' + id(a) + ': ' + a.message",
-        map[string]interface{}{"message": "Hello, World!"})
-    // In face of driver native errors, make sure to return them directly.
-    // Depending on the error, the driver may try to execute the function again.
-    if err != nil {
-        return nil, err
-    }
-    record, err := records.Single()
-    if err != nil {
-        return nil, err
-    }
-    // You can also retrieve values by name, with e.g. `id, found := record.Get("n.id")`
-    return &Item{
-        Message: record.Values[0].(string),
-    }, nil
+If you want to pass a map of properties to the query, you can do it like this:
+
+```go
+_, err = neo4j.ExecuteQuery(ctx, driver, "CREATE (n:Technology{name: $name, type: $type})", map[string]any{"name": "Memgraph", "type": "graph database"}, neo4j.EagerResultTransformer, neo4j.ExecuteQueryWithDatabase(""))
+if err != nil {
+	panic(err)
 }
+```
+This way your queries do not contain hard-coded values, and you can make your queries more dynamic.
 
-type Item struct {
-    Message string
+### Process the results
+
+In order to read results and serve them back to you Go application you need to handle them. Go is statically typed language and proper types need to be handled. 
+Depending on the type of the request you made, you can get different results back. Lets go over a few basic examples on how to handle different types and access properties of the returned results.
```

Comment:
```suggestion
Processing results from the database is important since we do not want to lose any data during conversions. To properly read results and serve them back to the Go application, they need to be cast into proper types. 

Depending on the type of request made, you can receive different results, `Nodes`, `Relationships`, `Paths` etc. Let's go over a few basic examples of how to handle different types and access properties of the returned results.
```